test/DSROracle.t.sol

```diff
@@ -0,0 +1,198 @@
+        pot.setChi(1.03e27);
+        oracle.refresh();
+
+        assertEq(oracle.getConversionRate(), 1.03e27);
+        assertEq(oracle.getConversionRate(block.timestamp + 365 days), 1.081499999999999999959902249e27);   // 5% interest on 1.03 value = 1.0815
+    }
+
+    function test_gas_getConversionRate_1hour() public {
+        vm.pauseGasMetering();
+        pot.setDSR(FIVE_PCT_APY_DSR);
+        pot.setChi(1.03e27);
+        oracle.refresh();
+        vm.resumeGasMetering();
+
+        oracle.getConversionRate(block.timestamp + 1 hours);
+    }
+
+    function test_gas_getConversionRate_1year() public {
+        vm.pauseGasMetering();
+        pot.setDSR(FIVE_PCT_APY_DSR);
+        pot.setChi(1.03e27);
+        oracle.refresh();
+        vm.resumeGasMetering();
+
+        oracle.getConversionRate(block.timestamp + 365 days);
+    }
+
+    function test_getConversionRate_pastRevert() public {
+        vm.expectRevert();
+        oracle.getConversionRate(block.timestamp - 365 days);
+    }
+
+    function test_getConversionRateBinomialApprox() public {
+        assertEq(oracle.getConversionRateBinomialApprox(), 1e27);
+        assertEq(oracle.getConversionRateBinomialApprox(block.timestamp + 365 days), 1e27);
+
+        pot.setDSR(FIVE_PCT_APY_DSR);
+        pot.setChi(1.03e27);
+        oracle.refresh();
+
+        assertEq(oracle.getConversionRateBinomialApprox(), 1.03e27);
+        assertEq(oracle.getConversionRateBinomialApprox(block.timestamp + 365 days), 1.081495968383924399665215760e27);   // 5% interest on 1.03 value = 1.0815
+    }
+
+    function test_gas_getConversionRateBinomialApprox_1hour() public {
+        vm.pauseGasMetering();
+        pot.setDSR(FIVE_PCT_APY_DSR);
+        pot.setChi(1.03e27);
+        oracle.refresh();
+        vm.resumeGasMetering();
+
+        oracle.getConversionRateBinomialApprox(block.timestamp + 1 hours);
+    }
+
+    function test_gas_getConversionRateBinomialApprox_1year() public {
+        vm.pauseGasMetering();
+        pot.setDSR(FIVE_PCT_APY_DSR);
+        pot.setChi(1.03e27);
+        oracle.refresh();
+        vm.resumeGasMetering();
+
+        oracle.getConversionRateBinomialApprox(block.timestamp + 365 days);
+    }
+
+    function test_getConversionRateBinomialApprox_pastRevert() public {
+        vm.expectRevert();
+        oracle.getConversionRateBinomialApprox(block.timestamp - 365 days);
+    }
+
+    function test_getConversionRateLinearApprox() public {
+        assertEq(oracle.getConversionRateLinearApprox(), 1e27);
+        assertEq(oracle.getConversionRateLinearApprox(block.timestamp + 365 days), 1e27);
+
+        pot.setDSR(FIVE_PCT_APY_DSR);
+        pot.setChi(1.03e27);
+        oracle.refresh();
+
+        assertEq(oracle.getConversionRateLinearApprox(), 1.03e27);
+        assertEq(oracle.getConversionRateLinearApprox(block.timestamp + 365 days), 1.078790164207174267760128000e27);   // 5% interest on 1.03 value = 1.0815, but linear approx is 1.0788
```

Comment:
Used APR value and got this result,

```
> 1.03 * 1.048790164207174267760128000
1.0802538691333894
```

How did you get the `1.078790164207174267760128000e27` value?

Reply:
Oh whoops the formula was wrong. Fixed.